code/io/timer.cpp

```diff
@@ -180,17 +180,23 @@ UI_TIMESTAMP ui_timestamp() {
 
 TIMESTAMP timestamp_delta(TIMESTAMP stamp, int delta_ms)
 {
-	if (!stamp.isValid() || stamp.isImmediate() || stamp.isNever())
+	if (!stamp.isValid() || stamp.isNever())
```

Comment:
This could be replaced with the new function:
```suggestion
	if (!stamp.isFinite())
```